### PR TITLE
Tweak icon component

### PR DESCRIPTION
**Pull request by amyrlam**
- Adjust icon code
  - Ref: Implementation trial in hashicorp/design-system-playground-fastboot#1 (icons would not display in Fastboot without this change)
- Update README 
  - Fix content
  - Switched to markdown

IS: 
image

Resolves: https://github.com/hashicorp/flight/issues/13

***

Questions: 
- Do we want a default size? I would be cool with that being 16. (update: can discuss with MS at pairing, seems like we wanted that, but needed to change syntax to work with fastboot. i think it can be done another way also)
- Should we change `fillColor` to `color`? `@isInlineBlock` sounds great to me. (update: https://github.com/hashicorp/flight/pull/42)

### CODE REVIEW

File: ember-flight-icons/README.md [hunk 1]
```diff
@@ -10,36 +9,38 @@ Goals:
...
+### Required usage
+
 The `name` must be specified.
 
 ```hbs
 <FlightIcon @name="activity" />
 ```
 
-Icons are 24x24 by default. To use the 16x16 size instead, set `size` to `small`:
+The `size` must be specified. Set the icon size by the number, e.g. 16 or 24.
```
[the remark this answers is not in the document]
Comment (amyrlam) [reply ember-flight-icons/README.md]: melanie is ok with this

**Comment by amyrlam** [comment]
got this locally working via instructions on commit message https://github.com/hashicorp/design-system-playground-fastboot/pull/3, ready for re-review

File: ember-flight-icons/README.md [hunk 2]
```diff
@@ -1,43 +1,44 @@
...
 ```
 
-Usage
-------------------------------------------------------------------------------
-The `name` must be specified.
+## Usage
+
+### Required usage
+
+The `name` must be specified. A default icon size of 16 will be provided.
```
Comment (MelSumner) [inline ember-flight-icons/README.md:30]:
```suggestion
The `name` must be specified. A default icon size of 24 will be provided.
```
24x24 is the default size.

File: ember-flight-icons/README.md [hunk 3]
```diff
@@ -1,43 +1,44 @@
...
 
 ```hbs
 <FlightIcon @name="activity" />
 ```
 
-Icons are 24x24 by default. To use the 16x16 size instead, set `size` to `small`:
+### Optional usage
+
+Icons are set to a `viewBox` square size of 16 by default. Right, now, we support a `viewBox` square size of 16 or 24. Pass in `size` to use the larger 24.
```
Comment (MelSumner) [inline ember-flight-icons/README.md:38]:
```suggestion
Icons are set to a `viewBox` square size of 24 by default. Right, now, we support a `viewBox` square size of 16 or 24. Pass in `size` to use the smaller 16 size.
```

File: ember-flight-icons/README.md [hunk 4]
```diff
@@ -1,43 +1,44 @@
...
 
 ```hbs
-<FlightIcon @name="zap" @size="small" />
+<FlightIcon @name="activity" @size="24" />
```
Comment (MelSumner) [inline ember-flight-icons/README.md:41]:
```suggestion
<FlightIcon @name="activity" @size="16" />
```

File: ember-flight-icons/README.md [hunk 5]
```diff
@@ -49,11 +50,14 @@ Icons are set to `display: inline-block` by default. To remove this, set `isInli
 Icons are set to `fill="currentColor"` by default. To change this, set the desired color:
 
 ```hbs
-<FlightIcon @name="activity" @fillColor="red" />
+<FlightIcon @name="activity" @size="24" @color="red" />
```
Comment (MelSumner) [inline ember-flight-icons/README.md:53]:
```suggestion
<FlightIcon @name="activity" @color="red" />
```

File: ember-flight-icons/README.md [hunk 6]
```diff
@@ -49,11 +50,14 @@ Icons are set to `display: inline-block` by default. To remove this, set `isInli
...
 ```
 
-Accessibility
-------------------------------------------------------------------------------
+### Note
+
+The `id` is generated by the component to be unique.
```
Comment (MelSumner) [inline ember-flight-icons/README.md:58]:
```suggestion
The `id` is automatically generated by the component, and will be unique.
```

File: ember-flight-icons/addon/components/flight-icon.hbs [hunk 7]
```diff
@@ -2,12 +2,13 @@
...
+  fill={{@color}} 
   height="{{this.size}}"
   id={{this.iconId}}
   viewBox="0 0 {{this.size}} {{this.size}}"
   width="{{this.size}}" 
   xmlns="http://www.w3.org/2000/svg"
 >
-  <use href='{{this.contextRootURL}}ember-flight-icons/icons/sprite.svg#{{@name}}-{{this.size}}'></use>
+
+  <use href='{{this.contextRootURL}}ember-flight-icons/icons/sprite.svg#{{@name}}-{{@size}}'></use>
```
Comment (MelSumner) [inline ember-flight-icons/addon/components/flight-icon.hbs:13]:
```suggestion
  <use href='{{this.contextRootURL}}ember-flight-icons/icons/sprite.svg#{{@name}}-{{this.size}}'></use>
```

File: ember-flight-icons/tests/dummy/app/templates/application.hbs [hunk 8]
```diff
@@ -6,10 +6,10 @@
   <h2>Example:</h2>
 <ul>
   <li>
-    This should be a large "activity" icon: <FlightIcon @name="activity" />
+    This should be a large "activity" icon: <FlightIcon @name="activity" @size="24" />
   </li>
   <li>This should be a small "activity" icon: <FlightIcon @name="activity" @size="16" /></li>
   <li>This should be a small "zap" icon: <FlightIcon @name="zap" @size="16" @isInlineBlock={{false}} /></li>
-  <li>This should be a red large "activity" icon: <FlightIcon @name="activity" @fillColor="red" /></li>
+  <li>This should be a red large "activity" icon: <FlightIcon @name="activity" @size="24" @color="red" /></li>
```
Comment (MelSumner) [inline ember-flight-icons/tests/dummy/app/templates/application.hbs:13]:
```suggestion
  <li>This should be a red large "activity" icon: <FlightIcon @name="activity" @color="red" /></li>
```

File: ember-flight-icons/addon/components/flight-icon.js [hunk 9]
```diff
@@ -9,13 +9,13 @@ export default class FlightIconComponent extends Component {
...
+   * Sets the color for the SVG
    *
-   * @param fillColor {string}
+   * @param color {string}
    * @default 'currentColor'
    */
-  get fillColor() {
-    return this.args.fillColor ?? 'currentColor';
+  get color() {
+    return this.args.color ?? 'currentColor';
```
Comment (MelSumner) [inline ember-flight-icons/addon/components/flight-icon.js:18]: This resolves the color TODO- color is set to `currentColor` as the default.

**Review by MelSumner — CHANGES_REQUESTED** [review]
@amyrlam requested a few changes to tighten things up, looking good!

File: ember-flight-icons/README.md (hunk 2 above)
Comment (amyrlam) [reply ember-flight-icons/README.md:30]: oh gotcha, i assumed it was the other way around. any reason why? curious if more icons are 16 vs 24 rn. but, let's address that separately, i'll make a triage issue

File: ember-flight-icons/addon/components/flight-icon.js (hunk 9 above)
Comment (amyrlam) [reply ember-flight-icons/addon/components/flight-icon.js:18]: i think the TODO is to add a test to check that / moved to https://github.com/hashicorp/flight/issues/54, so let's address that separately

**Comment by amyrlam** [comment]
made the changes and rebased, please review and merge if g2g!

**Review by MelSumner — APPROVED** [review]
Looks good, thank you Amy!

**Pull request merged**